backend/Squirrel.Core/Squirrel.Core.WebAPI/Extentions/ServiceCollectionExtensions.cs

```diff
@@ -1,23 +1,40 @@
-﻿using Squirrel.Core.BLL.MappingProfiles;
+﻿using AutoMapper;
+using FluentValidation.AspNetCore;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Options;
+using Squirrel.Core.BLL.Interfaces;
+using Squirrel.Core.BLL.MappingProfiles;
 using Squirrel.Core.BLL.Services;
+using Squirrel.Core.Common.Models;
 using Squirrel.Core.DAL.Context;
-using Squirrel.Core.BLL.Interfaces;
+using Squirrel.Core.DAL.Entities;
 using Squirrel.Core.WebAPI.Validators;
-using FluentValidation.AspNetCore;
-using Microsoft.EntityFrameworkCore;
 using System.Reflection;
 
 namespace Squirrel.Core.WebAPI.Extentions
 {
     public static class ServiceCollectionExtensions
     {
-        public static void RegisterCustomServices(this IServiceCollection services)
+        public static void RegisterCustomServices(this IServiceCollection services, IConfiguration configuration)
         {
             services
                 .AddControllers()
                 .AddNewtonsoftJson(options => options.SerializerSettings.ReferenceLoopHandling = Newtonsoft.Json.ReferenceLoopHandling.Ignore);
 
             services.AddTransient<ISampleService, SampleService>();
+
+            services.Configure<MongoDatabaseConnectionSettings>(
+                configuration.GetSection("MongoDatabase"));
+
+            services.AddTransient(typeof(IMongoService<Sample>), serviceProvider =>
+            {
+                var context = serviceProvider.GetRequiredService<SquirrelCoreContext>();
+                var mapper = serviceProvider.GetRequiredService<IMapper>();
+                var mongoDatabaseSettings = serviceProvider.GetRequiredService<IOptions<MongoDatabaseConnectionSettings>>();
+                string collectionName = "SampleCollection";
```

Comment:
can we get `string collectionName = "SampleCollection";` from settings instead of hardcode?

Reply:
Initially, I approached it that way, but Anatolii pointed out that using this approach would limit us from using it with more than one collection: "so it will be possible to define collection name in DI (As currently it's assigned via constructor and it's impossible to work with 2 collections)"

Reply:
so leave it as is?

Reply:
I currently believe that it's not such a bad approach. Additionally, this sample collection has been added here for demonstrative purposes. I will explore better ways when MongoDB will actually be used in our project.

Reply:
yeah, let it be